### PR TITLE
Add and improve documentation for dimensionality reduction

### PR DESCRIPTION
**Pull request by 900miles**
Improves documentation for the dimensionality reduction function, and adds a jupyter notebook tutorial for the same. I'm not the most familiar with the details of dimensionality reduction, so extra attention to the evaluation metrics / model summary section of the documentation would be appreciated! This is mostly what I could gather from limited information available online.

### CODE REVIEW

File: src/senselab/utils/tasks/dimensionality_reduction.py [hunk 1]
```diff
@@ -1,4 +1,44 @@
-"""This module provides the implementation of dimensionality reduction."""
+"""This module provides the implementation of dimensionality reduction.
+
+Dimensionality reduction is a crucial preprocessing step in many machine learning and data analysis pipelines.
+It aims to reduce the number of features in a high-dimensional dataset while preserving as much of the significant
```
Comment (fabiocat93) [inline src/senselab/utils/tasks/dimensionality_reduction.py]: i think that this is misleading. it's not just that you reduce the number of features (you could do that also by selecting a subset of features from the total number), but you do project your features into a lower dimensional space

File: src/senselab/utils/tasks/dimensionality_reduction.py [hunk 2]
```diff
@@ -1,4 +1,44 @@
...
+limited to:
+- **Reconstruction Error**: Measures the difference between the original data and its reduced representation when
+    projected back  to the original space. Commonly used for techniques like PCA.
+- **Trustworthiness**: Assesses how much neighbors in the high-dimensional space are also neighbors in the
+    low-dimensional space.
+- **Continuity**: Evaluates how well the local structure of the data is preserved in the reduced space.
+
+### Model summaries
+- **PCA**: Widely recognized for its effectiveness in linear dimensionality reduction and its ability to capture the
+    most variance in the data. A classic benchmark is the MNIST dataset, where PCA can reduce dimensions with minimal
```
Comment (fabiocat93) [inline src/senselab/utils/tasks/dimensionality_reduction.py]: i would remove the sentence about the MNIST dataset. you can apply pca to any dataset

File: src/senselab/utils/tasks/dimensionality_reduction.py [hunk 3]
```diff
@@ -1,4 +1,44 @@
...
+### Evaluation Metrics
+The effectiveness of dimensionality reduction techniques can be evaluated using several metrics, including but not
+limited to:
+- **Reconstruction Error**: Measures the difference between the original data and its reduced representation when
+    projected back  to the original space. Commonly used for techniques like PCA.
+- **Trustworthiness**: Assesses how much neighbors in the high-dimensional space are also neighbors in the
+    low-dimensional space.
+- **Continuity**: Evaluates how well the local structure of the data is preserved in the reduced space.
+
+### Model summaries
```
Comment (fabiocat93) [inline src/senselab/utils/tasks/dimensionality_reduction.py]: how do these methods differ? what info do they preserve or filter out? when is it better to use one method or the other?

File: tutorials/dimensionality_reduction_tutorial.ipynb
Comment (fabiocat93) [inline tutorials/dimensionality_reduction_tutorial.ipynb:1]: can you please add the colab button?

File: src/senselab/utils/tasks/dimensionality_reduction.py (hunk 1 above)
Comment (900miles) [reply src/senselab/utils/tasks/dimensionality_reduction.py]: Would this be better?

"It aims to project a high-dimensional dataset into a lower-dimensional subspace while preserving as much of the significant structure and information as possible."

File: src/senselab/utils/tasks/dimensionality_reduction.py (hunk 3 above)
Comment (900miles) [reply src/senselab/utils/tasks/dimensionality_reduction.py]: Which methods are you referring to?

Comment (fabiocat93) [reply src/senselab/utils/tasks/dimensionality_reduction.py]: PCA, t-SNE, UMAP

File: src/senselab/utils/tasks/dimensionality_reduction.py [hunk 4]
```diff
@@ -1,4 +1,60 @@
...
+### Evaluation Metrics
+The effectiveness of dimensionality reduction techniques can be evaluated using several metrics, including but not
+limited to:
+- **Reconstruction Error**: Measures the difference between the original data and its reduced representation when
+    projected back to the original space. Commonly used for techniques like PCA.
+- **Trustworthiness**: Assesses how much the model preserves neighbor relationships between the high-dimensional
+    space and the low-dimensional space.
+- **Continuity**: Evaluates how well the local structure of the data is preserved in the reduced space.
+
+### Supported models
```
Comment (fabiocat93) [inline src/senselab/utils/tasks/dimensionality_reduction.py:25]: way better, would be great to have some refs

Comment (900miles) [reply src/senselab/utils/tasks/dimensionality_reduction.py:25]: Sounds good, I've added links to the t-SNE and UMAP papers, as well as a Nature primer on PCA that I found.

**Comment by 900miles** [comment]
Hi @fabiocat93, is there anything else that is missing before this can be merged?

**Comment by fabiocat93** [comment]
> Hi @fabiocat93, is there anything else that is missing before this can be merged?

you are good!

**Review by fabiocat93 — APPROVED** [review]

**Pull request merged**